clang/lib/Format/WhitespaceManager.cpp

```diff
@@ -1316,6 +1316,8 @@ void WhitespaceManager::alignArrayInitializersRightJustified(
         auto Offset = std::distance(Cells.begin(), CellIter);
         for (const auto *Next = CellIter->NextColumnElement; Next;
              Next = Next->NextColumnElement) {
+          if (RowCount >= CellDescs.CellCounts.size())
+            break;
```

Comment:
```suggestion
        for (const auto *Next = CellIter->NextColumnElement; Next && RowCount < CellDescs.CellCounts.size();
             Next = Next->NextColumnElement, ++RowCount) {
```
Maybe?

Then the `++RowCount` below must be removed.

Reply:
I intentionally made it the same as lines 1384-1385 below so that lines 1315-1323 would remain copy-pasted from lines 1380-1388. This will make future refactoring easier.

Reply:
I would change the other lines too. I think it's better to see the condition in the `for`, but it's your call.

Reply:
The same/similar code appeared twice in this file and once in the header file. If I were to refactor it, I might not use the same for-loop, so I'll leave it as is for now. (There are also much more cleanup that can be done with this option, but I don't know if it's worthwhile as it's so buggy even after @mydeveloperday limited it to matrices.)